spec/features/users/google_login_spec.rb

```diff
@@ -0,0 +1,108 @@
+require "rails_helper"
+
+describe "User logs in through facebook" do
+
+  it "logs in through registration page" do
+        visit "/"
+        click_on "Sign Up"
+        expect(current_path).to eq("/users/new")
+        click_link('Sign Up with Google')
+
+        expect(current_path).to eq("/profile")
+        expect(page).to have_content("You are now logged in as John")
+
+        within("#logout") do
+            click_link "Logout"
+        end
+
+        expect(current_path).to eq("/")
+        expect(page).to have_content("You are now logged out")
+        expect(page).to have_content("Login")
+        expect(page).not_to have_content("Logout")
+    end
+
+    it "logs in through login page" do
+        visit "/"
+        click_on "Login"
+        expect(current_path).to eq("/login")
+
+        click_link('Sign in with Google')
+
+        expect(current_path).to eq("/profile")
+        expect(page).to have_content("You are now logged in as John")
+        within("#logout") do
+            click_link "Logout"
+        end
+
+        expect(current_path).to eq("/")
+        expect(page).to have_content("You are now logged out")
+        expect(page).to have_content("Login")
+        expect(page).not_to have_content("Logout")
+
+    end
+
+    it "logs in when user already has an account" do
+      visit "/"
+      click_on "Sign Up"
+      click_link('Sign Up with Google')
+      click_link 'Logout'
+      click_on 'Login'
+      click_link('Sign in with Google')
+      expect(page).to have_content("You are now logged in as John")
+    end
+
+    it "has a sad path if the facebook login fails" do
+      OmniAuth.config.mock_auth[:google_oauth2] = OmniAuth::AuthHash.new({
+        "provider" => "google_oauth2",
+        "uid" => "100000000000000000000",
+        "info" => {
+            "name" => "John Smith",
+            "email" => nil,
```

Comment:
Sad path test if we don't get an email in the oauth callback for whatever reason.

Reply:
Since this is in the rails helper file, do we need it here?

Reply:
I actually just included it in this specific test to make sure we have a sad path if something fails with the OAuth. The email on this one is `nil` and the one in the rails helper has an actual email.

Reply:
Oh I see, good thinking.